File: CS/ViewModels/MenuItemViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace DXSample.ViewModels {
    public class MenuItemViewModel : ViewModelBase {
        public MenuItemViewModel() {
            Items = new ObservableCollection<MenuItemViewModel>();
        }

        public string Title {
            get { return GetProperty(() => Title); }
            set { SetProperty(() => Title, value); }
        }

        public ObservableCollection<MenuItemViewModel> Items {
            get { return GetProperty(() => Items); }
            set { SetProperty(() => Items, value); }
        }
    }
}

File: VB/ViewModels/MenuItemViewModel.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace DXSample.ViewModels

    Public Class MenuItemViewModel
        Inherits ViewModelBase

        Public Sub New()
            Items = New ObservableCollection(Of MenuItemViewModel)()
        End Sub

        Public Property Title As String
            Get
                Return GetProperty(Function() Me.Title)
            End Get

            Set(ByVal value As String)
                SetProperty(Function() Title, value)
            End Set
        End Property

        Public Property Items As ObservableCollection(Of MenuItemViewModel)
            Get
                Return GetProperty(Function() Me.Items)
            End Get

            Set(ByVal value As ObservableCollection(Of MenuItemViewModel))
                SetProperty(Function() Items, value)
            End Set
        End Property
    End Class
End Namespace
